contrib/localnet/orchestrator/smoketest/test_deposit_eth.go

```diff
@@ -53,6 +74,67 @@ func (sm *SmokeTest) TestDepositEtherIntoZRC20() {
 	fmt.Printf("  value: %d\n", signedTx.Value())
 	fmt.Printf("  block num: %d\n", receipt.BlockNumber)
 
+	{
+		LoudPrintf("Merkle Proof\n")
+		txHash := receipt.TxHash
+		blockHash := receipt.BlockHash
+		txIndex := int(receipt.TransactionIndex)
+
+		block, err := sm.goerliClient.BlockByHash(context.Background(), blockHash)
+		if err != nil {
+			panic(err)
+		}
+		i := 0
+		for {
+			if i > 20 {
+				panic("block header not found")
+			}
+			_, err := sm.observerClient.GetBlockHeaderByHash(context.Background(), &observertypes.QueryGetBlockHeaderByHashRequest{
+				BlockHash: blockHash.Bytes(),
+			})
+			if err != nil {
+				fmt.Printf("WARN: block header not found; retrying...\n")
+				time.Sleep(2 * time.Second)
+			} else {
+				fmt.Printf("OK: block header found\n")
+				break
+			}
+			i++
+		}
+
+		trie := ethereum2.NewTrie(block.Transactions())
+		if trie.Hash() != block.Header().TxHash {
+			panic("tx root hash & block tx root mismatch") // FIXME: don't do this in production
+		}
+		txProof, err := trie.GenerateProof(txIndex)
+		if err != nil {
+			panic("error generating txProof") // FIXME: don't do this in production
+		}
+		val, err := txProof.Verify(block.TxHash(), txIndex)
+		if err != nil {
+			panic("error verifying txProof") // FIXME: don't do this in production
+		}
+		var txx ethtypes.Transaction
+		err = txx.UnmarshalBinary(val)
+		if err != nil {
+			panic("error unmarshalling txProof'd tx") // FIXME: don't do this in production
```

Comment:
What does the `FIXME` means? we always use `panic` in smoketest isn't it?

Reply:
removed it; that was copied from somewhere else.